### PR TITLE
Added .Net Core Project Test Runner Strategy

### PR DESCRIPTION
**Pull request by gogo4ds**
Implement a TestRunner strategy for .NET Core Projects #2857
Closes https://github.com/SoftUni-Internal/suls-issues/issues/2857

### CODE REVIEW

File: Open Judge System/OJS.Common/Models/ExecutionStrategyType.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ public enum ExecutionStrategyType
         CSharpProjectTestsExecutionStrategy = 24,
         CSharpAspProjectTestsExecutionStrategy = 28,
         CSharpPerformanceProjectTestsExecutionStrategy = 31,
-        CSharpTestRunner = 10,
+        DotNetTestRunner = 10,
```
Comment (Innos) [inline Open Judge System/OJS.Common/Models/ExecutionStrategyType.cs]: Rename to DotNetCoreTestRunner

File: Open Judge System/Data/OJS.Data/Migrations/DefaultMigrationConfiguration.cs [hunk 2]
```diff
@@ -197,12 +197,11 @@ protected void SeedSubmissionTypes(OjsDbContext context)
...
-                    Name = "C# test runner",
-                    CompilerType = CompilerType.MsBuild,
-                    AdditionalCompilerArguments =
-                        "/t:rebuild /p:Configuration=Release,Optimize=true /verbosity:quiet /nologo",
+                    Name = ".NET Core Test Runner",
+                    CompilerType = CompilerType.DotNetCompiler,
+                    AdditionalCompilerArguments = string.Empty,
                     ExecutionStrategyType =
-                        ExecutionStrategyType.CSharpTestRunner,
+                        ExecutionStrategyType.DotNetCoreTestRunner,
```
Comment (YanaSlavcheva) [inline Open Judge System/Data/OJS.Data/Migrations/DefaultMigrationConfiguration.cs]: why new line?

File: Open Judge System/OJS.Common/Extensions/FileHelpers.cs [hunk 3]
```diff
@@ -82,6 +82,19 @@ public static string FindFileMatchingPattern(string workingDirectory, string pat
...
+            Func<string, TOut> orderBy)
+        {
+            var files = DiscoverAllFilesMatchingPattern(workingDirectory, pattern);
+
+            string discoveredFile = files.Where(where).OrderByDescending(orderBy).First();
+
+            return ProcessModulePath(discoveredFile);
+        }
+
         public static void AddFilesToZipArchive(string archivePath, string pathInArchive, params string[] filePaths)
```
Comment (YanaSlavcheva) [inline Open Judge System/OJS.Common/Extensions/FileHelpers.cs]: split row maybe?

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/DotNetCoreTestRunnerExecutionStrategy.cs [hunk 4]
```diff
@@ -104,26 +129,54 @@ public static void Main()
...
+
             if (!compileResult.IsCompiledSuccessfully)
             {
                 return result;
             }
 
-            var outputAssemblyPath = this.PreprocessAndCompileTestRunner(executionContext, compileResult.OutputFile);
+            result.IsCompiledSuccessfully = compileResult.IsCompiledSuccessfully;
+
+            var outputAssemblyPath = this.PreprocessAndCompileTestRunner(executionContext, Path.GetDirectoryName(compileResult.OutputFile));
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/DotNetCoreTestRunnerExecutionStrategy.cs]: split

File: Open Judge System/Workers/OJS.Workers.ExecutionStrategies/DotNetCoreTestRunnerExecutionStrategy.cs [hunk 5]
```diff
@@ -150,7 +203,8 @@ private string PreprocessAndCompileTestRunner(ExecutionContext executionContext,
             compilerParameters.ReferencedAssemblies.Add("mscorlib.dll");
             compilerParameters.ReferencedAssemblies.Add("System.dll");
             compilerParameters.ReferencedAssemblies.Add("System.Core.dll");
-            compilerParameters.ReferencedAssemblies.AddRange(Directory.GetFiles(outputDirectory).Where(f => f.EndsWith(".dll") || f.EndsWith(".exe")).ToArray());
+            var referencedTypes = Directory.GetFiles(outputDirectory).Where(f => f.EndsWith(".dll") || f.EndsWith(".exe")).ToArray();
```
Comment (YanaSlavcheva) [inline Open Judge System/Workers/OJS.Workers.ExecutionStrategies/DotNetCoreTestRunnerExecutionStrategy.cs]: split; also strings in constants maybe?

File: Open Judge System/OJS.Common/Extensions/FileHelpers.cs (hunk 3 above)
Comment (gogo4ds) [reply Open Judge System/OJS.Common/Extensions/FileHelpers.cs]: It is less than 120 characters long (117)

File: Open Judge System/OJS.Common/Extensions/FileHelpers.cs [hunk 6]
```diff
@@ -82,6 +82,19 @@ public static string FindFileMatchingPattern(string workingDirectory, string pat
...
 
+        public static string FindFileMatchingPattern<TOut>(
+            string workingDirectory,
+            string pattern,
+            Func<string, bool> where,
+            Func<string, TOut> orderBy)
+        {
+            var files = DiscoverAllFilesMatchingPattern(workingDirectory, pattern);
+
+            string discoveredFile = files.Where(where).OrderByDescending(orderBy).First();
```
Comment (YanaSlavcheva) [inline Open Judge System/OJS.Common/Extensions/FileHelpers.cs]: var

**Pull request merged**